src/openloco/windows/toolbar_top.cpp

```diff
@@ -0,0 +1,1246 @@
+#include "../graphics/image_ids.h"
+#include "../input.h"
+#include "../interop/interop.hpp"
+#include "../localisation/string_ids.h"
+#include "../objects/interface_skin_object.h"
+#include "../objects/land_object.h"
+#include "../objects/objectmgr.h"
+#include "../objects/road_object.h"
+#include "../objects/track_object.h"
+#include "../objects/water_object.h"
+#include "../stationmgr.h"
+#include "../things/thingmgr.h"
+#include "../things/vehicle.h"
+#include "../townmgr.h"
+#include "../ui/WindowManager.h"
+#include "../ui/dropdown.h"
+
+using namespace openloco::interop;
+
+namespace openloco::ui::windows::toolbar_top
+{
+    namespace widx
+    {
+        enum
+        {
+            loadsave_menu,
+            audio_menu,
+            zoom_menu,
+            rotate_menu,
+            view_menu,
+
+            terraform_menu,
+            railroad_menu,
+            road_menu,
+            port_menu,
+            build_vehicles_menu,
+
+            vehicles_menu,
+            stations_menu,
+            towns_menu,
+        };
+    }
+
+    static widget_t _widgets[] = {
+        make_widget({ 0, 0 }, { 30, 28 }, widget_type::wt_7, 0),
+        make_widget({ 30, 0 }, { 30, 28 }, widget_type::wt_7, 0),
+        make_widget({ 74, 0 }, { 30, 28 }, widget_type::wt_7, 1),
+        make_widget({ 104, 0 }, { 30, 28 }, widget_type::wt_7, 1),
+        make_widget({ 134, 0 }, { 30, 28 }, widget_type::wt_7, 1),
+
+        make_widget({ 267, 0 }, { 30, 28 }, widget_type::wt_7, 2),
+        make_widget({ 387, 0 }, { 30, 28 }, widget_type::wt_7, 2),
+        make_widget({ 357, 0 }, { 30, 28 }, widget_type::wt_7, 2),
+        make_widget({ 417, 0 }, { 30, 28 }, widget_type::wt_7, 2),
+        make_widget({ 417, 0 }, { 30, 28 }, widget_type::wt_7, 2),
+
+        make_widget({ 490, 0 }, { 30, 28 }, widget_type::wt_7, 3),
+        make_widget({ 520, 0 }, { 30, 28 }, widget_type::wt_7, 3),
+        make_widget({ 460, 0 }, { 30, 28 }, widget_type::wt_7, 3),
+        widget_end(),
+    };
+
+    static window_event_list _events;
+
+    static loco_global<uint8_t[40], 0x00113DB20> menu_options;
+
+    static loco_global<uint8_t, 0x00525FAA> last_railroad_option;
+    static loco_global<uint8_t, 0x00525FAB> last_road_option;
+    static loco_global<uint8_t, 0x00525FAF> last_vehicles_option;
+    static loco_global<uint8_t, 0x0052622C> last_build_vehicles_option;
+
+    static loco_global<uint8_t, 0x009C870C> last_town_option;
+    static loco_global<uint8_t, 0x009C870D> last_port_option;
+
+    static loco_global<int8_t[18], 0x0050A006> available_objects;
+
+    static void on_resize(window* window);
+    static void on_mouse_down(window* window, widget_index widgetIndex);
+    static void on_dropdown(window* window, widget_index widgetIndex, int16_t itemIndex);
+    static void on_update(window* window);
+    static void prepare_draw(window* window);
+    static void draw(window* window, gfx::drawpixelinfo_t* dpi);
+
+    // 0x00438A8F
```

Comment:
```suggestion
    // 0x00438B26
```

Reply:
Ah, right, `438A8F` is `gui::init` instead.